pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -273,17 +273,34 @@ protected void failPendingBatchReceives(ConcurrentLinkedQueue<OpBatchReceive<T>>
 
     abstract protected CompletableFuture<Messages<T>> internalBatchReceiveAsync();
 
+    private static void validateMessageId(Message<?> message) throws PulsarClientException {
+        if (message == null) {
+            throw new PulsarClientException.InvalidMessageException("Non-null message is required");
+        }
+        if (message.getMessageId() == null) {
+            throw new PulsarClientException.InvalidMessageException("Cannot handle message with null messageId");
+        }
+    }
+
+    private static void validateMessageId(MessageId messageId) throws PulsarClientException {
+        if (messageId == null) {
+            throw new PulsarClientException.InvalidMessageException("Cannot handle message with null messageId");
+        }
+    }
+
     @Override
     public void acknowledge(Message<?> message) throws PulsarClientException {
+        validateMessageId(message);
         try {
             acknowledge(message.getMessageId());
-        } catch (NullPointerException npe) {
-            throw new PulsarClientException.InvalidMessageException(npe.getMessage());
+        } catch (Exception e) {
```

Comment:
why this is not a `catch PulsarClientException ` ?
the same applies to other "catch Exception" clauses below

Reply:
@eolivelli  The method throws PulsarClientException / point of this catch it to wrap other exception types into the PulsarClientException. acknowledge() internally gets into acknowledgeAsync().get() which can throw InterruptedException/ExecutionException.

Reply:
in this case
we have to handle InterruptedException and call Thread.currentThread.interrupt();

Reply:
also if it is possible it is better to catch specific exception type

Reply:
@eolivelli fixed